.circleci/config.yml

```diff
@@ -29,12 +29,11 @@ jobs:
           echo "$WEBAPP_GIT_COMMIT"
 
           curl -f -o ./dist.tar.gz https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz
-          if [[ $? == 0 ]]
-          then
-            mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1
-          else
-            npm ci && make build
-          fi
+          mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1
```

Comment:
This is fine, but it is not an ideal fix. The problem with this approach is that if any of the commands fail, it will try to execute the on_fail step.

Reply:
OK, so I created the trap before the command so it only triggers once it reached the command.